nautobot/extras/models/approvals.py

```diff
@@ -41,19 +43,39 @@ class ApprovalWorkflowDefinition(PrimaryModel):
         default=dict,
         help_text="Constraints to filter the objects that can be approved using this workflow.",
     )
+    priority = models.IntegerField(
+        default=0, help_text="Determines workflow relevance when multiple apply. Lower values indicate higher priority."
+    )
     documentation_static_path = "docs/user-guide/platform-functionality/approval-workflow.html"
     is_dynamic_group_associable = False
 
     class Meta:
-        """Meta class for ApprovalWorkflow."""
+        """Meta class for ApprovalWorkflow Definition."""
 
         verbose_name = "Approval Workflow Definition"
         ordering = ["name"]
+        unique_together = [["model_content_type", "priority"]]
 
     def __str__(self):
         """Stringify instance."""
         return self.name
 
+    @classmethod
+    def find_for_model(cls, model_instance) -> Optional["ApprovalWorkflowDefinition"]:
```

Comment:
Might consider adding a custom Manager class for `ApprovalWorkflowDefinition` and implementing this method there instead - would be more consistent with existing APIs like `CustomField.objects.get_for_model` and the like.